main_app/igdb_api.py

```diff
@@ -0,0 +1,75 @@
+import requests
+from math import ceil
+from itechart_project.settings import IGDB_API_KEY
+
+
+class IGDB:
+    __api_key = IGDB_API_KEY
+    __api_url = 'https://api-v3.igdb.com/'
+    __headers = {
+        'user-key': __api_key,
+        'Accept': 'application/json'
+    }
+
+    def __init__(self):
+        self.__last_headers_x_count = 1
+
+    def __api_get(self, additional_string: str = ''):
+        data = requests.get(self.__api_url + additional_string, headers=self.__headers)
+        x_count = int(data.headers.get('X-Count'))
+        if x_count > 1:
+            self.__last_headers_x_count = x_count
+        return data.json()
+
+    def api_get_game(self, game_id):
+        data = self.__api_get(f'games/{game_id}?fields=rating,version_title,aggregated_rating,'
+                              f' screenshots,summary,cover,platforms,genres')
+        print(data)
+        return data
+
+    def __generate_filters(self, search: str = None, genres: str = None,
+                           platforms: str = None, user_ratings: tuple = None) -> str:
+        result = []
+        if search is not None:
+            result.append(f'&search={search.replace(" ", "%20")}&')
```

Comment:
I'm not sure but I thinkt that probably you can use standard `urllib` in your code, that would be more _pythonic_ way
```
import urllib
text_encoded = urllib.parse.quote(text)
```

Also take a look at this answer about urlencode
https://stackoverflow.com/a/40557716/2614426